lib/kafo/validator.rb

```diff
@@ -2,79 +2,111 @@
 module Kafo
   class Validator
 
-    def self.prepare_functions
-      return true if @reset || !Puppet::Parser::Functions.respond_to?(:reset)
-      Puppet::Parser::Functions.reset
-      @reset = true
+    def initialize
+      @logger = KafoConfigure.logger
     end
 
-    def initialize(params)
-      self.class.prepare_functions
-      definitions = KafoConfigure.module_dirs.map do |dir|
-        validate_files = dir + '/*/lib/puppet/parser/functions/validate_*.rb'
-        is_function_files = dir + '/*/lib/puppet/parser/functions/is_*.rb'
-        Dir.glob(validate_files) + Dir.glob(is_function_files)
-      end.flatten
-
-      definitions.each do |file|
-        require File.expand_path(file)
+    def validate_absolute_path(args)
+      args.each do |arg|
+        unless arg.to_s.start_with?('/')
+          @logger.error "#{arg.inspect} is not an absolute path."
+          return false
+        end
       end
+      return true
+    end
 
-      @params = params
-      @logger = KafoConfigure.logger
+    def validate_array(args)
+      args.each do |arg|
+        unless arg.is_a?(Array)
+          @logger.error "#{arg.inspect} is not a valid array."
+          return false
+        end
+      end
+      return true
+    end
 
-      @cache ||= Hash.new do |hash, key|
-        @logger.debug "Looked for #{key}"
-        param     = @params.select { |p| p.name == key.to_s }.first
-        hash[key] = param.nil? ? nil : param.value
+    def validate_bool(args)
+      args.each do |arg|
+        unless arg.is_a?(TrueClass) || arg.is_a?(FalseClass)
+          @logger.error "#{arg.inspect} is not a valid boolean."
+          return false
+        end
       end
+      return true
     end
 
-    def lookupvar(name, options = {})
-      @cache[name]
+    def validate_hash(args)
+      args.each do |arg|
+        unless arg.is_a?(Hash)
+          @logger.error "#{arg.inspect} is not a valid hash."
+          return false
+        end
+      end
+      return true
     end
 
-    # for puppet >= 3
-    def include?(value)
-      true
+    def validate_integer(args)
+      value = args[0]
+      max = args[1]
+      min = args[2]
+      int = Integer(value.to_s)
+      if min && int < min.to_i
+        @logger.error "#{value} must be at least #{min}."
+        return false
+      end
+      if max && int > max.to_i
+        @logger.error "#{value} must be less than #{max}."
+        return false
+      end
+      return true
+    rescue TypeError, ArgumentError
+      @logger.error "#{value.inspect} is not a valid integer."
+      return false
     end
 
-    # for puppet >= 3
-    def [](value, *args)
-      lookupvar(value)
+    def validate_listen_on(args)
```

Comment:
no, maybe just add a comment that this is not a standard validation, it's fine until we have a better way